src/Type-Guards/index.ts

```diff
@@ -0,0 +1,124 @@
+interface IBudget {
+  debit: number;
+  credit: number;
+}
+
+interface IDepartmentInfo {
+  name: string;
+  domain: string;
+}
+
+interface IDepartment {
+  name: string;
+  domain: string;
+  employees: IEmployee[];
+  budget: IBudget;
+  culculateBalance(): number;
+  addEmployee(employee: IEmployee): void;
+  removeEmployee(employeeId: number): void;
+  promoteToEmployee(preHired: IPreHiredEmployee): void;
+}
+
+interface IEmployee {
+  id: number;
+  firstName: string;
+  lastName: string;
+  paymentInfo: {
+    bankAccountNumber: string;
+    salary: number;
+  };
+  status: 'active' | 'inactive' | 'unpaid_leave';
+  department: IDepartmentInfo;
+}
+
+interface IPreHiredEmployee {
+  firstName: string;
+  lastName: string;
+  salary: number;
+  bankAccountNumber: string;
+}
+
+interface ICompany {
+  name: string;
+  departments: Department[];
+  preHiredEmployee: IPreHiredEmployee[];
+  allEmployees: IEmployee[];
```

Comment:
All employees, including both current and newly hired employees, should be included.